### PR TITLE
Add option to not receive reactions 

### PR DESCRIPTION
**Pull request by stelar7**

### CODE REVIEW

**Comment by Querijn** [comment]
Looks good, can you edit the welcome message to indicate they can opt out of reacts?

File: src/AutoReact.ts [hunk 1]
```diff
@@ -66,18 +88,23 @@ export default class AutoReact {
...
-            && !greeting.startsWith("hey ") && greeting != "hey"
-            && !greeting.startsWith("good morning ") && greeting != "good morning"
-            && !greeting.startsWith("goodmorning ") && greeting != "goodmorning"
-            && !greeting.startsWith("good evening ") && greeting != "good evening"
-            && !greeting.startsWith("goodevening ") && greeting != "goodevening"
-            && !greeting.startsWith("good night ") && greeting != "good night"
-            && !greeting.startsWith("goodnight ") && greeting != "goodnight"
-            && !greeting.startsWith("goodday ") && greeting != "goodday"
-            && !greeting.startsWith("good day ") && greeting != "good day")
+        const words: String[] = [
```
Comment (molenzwiebel) [inline src/AutoReact.ts]: `string`, not `String` ;)

File: src/AutoReact.ts [hunk 2]
```diff
@@ -66,18 +88,23 @@ export default class AutoReact {
...
+            "hello", "hi", "hey",
+            "good morning", "goodmorning",
+            "good evening", "goodevening",
+            "good night", "goodnight",
+            "good day", "goodday"
+        ];
+
+        const shouldReact = words.filter((x: string) => !greeting.startsWith(x)).filter((x: string) => greeting != x).length != words.length;
```
Comment (molenzwiebel) [inline src/AutoReact.ts]: You should be able to leave out the casts here if you correct the previous comments. Also, if I'm reading this correctly, it should be able to be simplified to just `words.some(x => greeting.startsWith(x))`, which looks a lot cleaner.

File: src/AutoReact.ts [hunk 3]
```diff
@@ -66,18 +88,23 @@ export default class AutoReact {
...
-            && !greeting.startsWith("hey ") && greeting != "hey"
-            && !greeting.startsWith("good morning ") && greeting != "good morning"
-            && !greeting.startsWith("goodmorning ") && greeting != "goodmorning"
-            && !greeting.startsWith("good evening ") && greeting != "good evening"
-            && !greeting.startsWith("goodevening ") && greeting != "goodevening"
-            && !greeting.startsWith("good night ") && greeting != "good night"
-            && !greeting.startsWith("goodnight ") && greeting != "goodnight"
-            && !greeting.startsWith("goodday ") && greeting != "goodday"
-            && !greeting.startsWith("good day ") && greeting != "good day")
+        const words: string[] = [
```
Comment (molenzwiebel) [inline src/AutoReact.ts]: You shouldn't even need the type specifier here, but I'll allow it 😛

**Comment by Querijn** [comment]
Original thinking was broken, seems like it got some toggle code of its own, but the reply of the bot was wrong.

**Pull request merged**